extensions/azure/data-plane-azure-storage/src/test/java/org/eclipse/dataspaceconnector/azure/dataplane/azurestorage/pipeline/AzureStorageDataSinkTest.java

```diff
@@ -0,0 +1,124 @@
+package org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline;
+
+import com.github.javafaker.Faker;
+import org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.adapter.BlobAdapter;
+import org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.adapter.BlobAdapterFactory;
+import org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.schema.AzureBlobStoreSchema;
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.DataSource.Part;
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.InputStreamDataSource;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataFlowRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.TestCustomException;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createAccountName;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createBlobName;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createContainerName;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createRequest;
+import static org.eclipse.dataspaceconnector.azure.dataplane.azurestorage.pipeline.AzureStorageTestFixtures.createSharedKey;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class AzureStorageDataSinkTest {
+
+    static Faker faker = new Faker();
```

Comment:
Do we need this dependency?

Reply:
It's quite a good practice to make test data variable so that we don't program against a test case, and also makes it clearer what test data is arbitrary vs special values. Faker was already used in EDC tests.

Reply:
Hmm, I didn't see it was included in the logger extension. For these tests, I don't see the value in pulling in another dependency for this but since it is already being used, let's just keep it.